demo/c/pv_speaker_demo.c

```diff
@@ -187,38 +187,49 @@ int main(int argc, char *argv[]) {
     }
 
     fprintf(stdout, "Playing audio...\n");
-    if (pcm) {
-        char *pcmData = (char *) pcm;
-        for (int i = 0; i < num_samples; i += frame_length) {
-            bool is_last_frame = i + frame_length >= num_samples;
+    if (pcm_data) {
+        int8_t *pcm = (int8_t *) pcm_data;
+        int32_t total_written_length = 0;
 
+        while (!is_interrupted && total_written_length < num_samples) {
+            int32_t written_length = 0;
             status = pv_speaker_write(
                     speaker,
-                    is_last_frame ? num_samples - i : frame_length,
-                    &pcmData[i * bits_per_sample / 8]);
+                    &pcm[total_written_length * bits_per_sample / 8],
+                    num_samples - total_written_length,
+                    &written_length);
             if (status != PV_SPEAKER_STATUS_SUCCESS) {
                 fprintf(stderr, "Failed to write with %s.\n", pv_speaker_status_to_string(status));
                 exit(1);
             }
-            if (is_interrupted) {
-                fprintf(stdout, "\nStopped audio...\n");
-                break;
-            }
+            total_written_length += written_length;
         }
 
         free(pcm);
     }
 
+    if (is_interrupted) {
+        fprintf(stdout, "\nStopped audio...\n");
+    } else {
+        fprintf(stdout, "Waiting for audio to finish...\n");
+        int32_t pcm_length = 0;
+        int16_t pcm[pcm_length];
+        int8_t *pcm_ptr = (int8_t *) pcm;
+        int32_t written_length = 0;
+        status = pv_speaker_flush(speaker, pcm_ptr, pcm_length, &written_length);
```

Comment:
flush should only take the `speaker` argument? specially here it doesn't seem other arguments do anything

Reply:
if you pass actual pcm to flush, it will also play it. So with orca, the flushed pcm will be passed here. Since with c there is no optional arguments, I thought it made sense to just pass a pointer to empty array - is there a better pattern/convention?